helm/app-operator-chart/templates/deployment.yaml

```diff
@@ -1,4 +1,4 @@
-apiVersion: apps/v1
+apiVersion: extensions/v1beta1
```

Comment:
The version has regressed. We should use `apps/v1`.

Reply:
I'm not sure why `apps/v1` gave an error, ```{"caller":"github.com/giantswarm/app-operator/vendor/github.com/giantswarm/backoff/notifier.go:13","level":"warning","message":"retrying backoff in '582.280027ms' due to error","stack":"[{/go/src/github.com/giantswarm/app-operator/vendor/github.com/giantswarm/helmclient/helmclient.go:488: } {rpc error: code = Unknown desc = release giantswarm-app-operator failed: Deployment.apps \"app-operator\" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{\"giantswarm.io/service-type\":\"managed\", \"app\":\"app-operator\"}: `selector` does not match template `labels`]}]","time":"2019-02-05T12:21:35.917182+00:00"}``` 
That's why I change this to `extnsions/v1beta`, any idea?

Reply:
`extensions/v1beta1` is an old API version for the deployment resource. We should move to `apps/v1` wherever possible and especially for new components.

The validation for `apps/v1` is a bit stricter so you need to add the YAML below. For the old API version this selector was optional.

```
  selector:
    matchLabels:
      app: {{ .Values.name }}
```